packages/flutter_tools/lib/src/asset.dart

```diff
@@ -22,9 +19,25 @@ class AssetBundle {
   static const String defaultManifestPath = 'pubspec.yaml';
   static const String _kAssetManifestJson = 'AssetManifest.json';
   static const String _kFontManifestJson = 'FontManifest.json';
-  static const String _kFontSetMaterial = 'material';
   static const String _kLICENSE = 'LICENSE';
 
+  static const String _materialIconsFontUpgradeInstructions = '''
+═══════════════════════════════════════════════════════════════════════════════════
+'uses-material-design' in pubspec.yaml is deprecated
+
+The flag is now replaced by a generic Dart package to provide the Material icons asset.
+
+Replace
```

Comment:
/cc @bwilkerson, @pq - an example of what would make a good lint for pubspec files

Reply:
Interesting!

If someone could capture the advice in a sentence (or more), I'll open a lint request.  Thanks!